src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBUtils.java

```diff
@@ -163,7 +192,59 @@
             //Rebuild full vidMap
             return updatedVidMapBuilder.build();
         }
-        return null;
+        return vidMapPB;
+    }
+
+    /**
+     * Update vid Protobuf object with a new variable length descriptor, as for allele-specific annotations
+     * @param vidMapPB input vid object
+     * @param fieldNameToIndexInVidFieldsList name to index in list
+     * @param fieldName INFO field name
+     * @param newCombineOperation combine op ("histogram_sum", "element_wise_float_sum", "strand_bias_table")
+     * @return updated vid Protobuf object if field exists, else null
+     */
+    public static GenomicsDBVidMapProto.VidMappingPB updateAlleleSpecificINFOFieldCombineOperation(
+            final GenomicsDBVidMapProto.VidMappingPB vidMapPB,
+            final Map<String, Integer> fieldNameToIndexInVidFieldsList,
+            final String fieldName,
+            final String newCombineOperation)
+    {
+        int fieldIdx = fieldNameToIndexInVidFieldsList.containsKey(fieldName)
+                ? fieldNameToIndexInVidFieldsList.get(fieldName) : -1;
+        if(fieldIdx >= 0) {
+            //Would need to rebuild vidMapPB - so get top level builder first
+            GenomicsDBVidMapProto.VidMappingPB.Builder updatedVidMapBuilder = vidMapPB.toBuilder();
+            //To update the list element corresponding to fieldName, we get the builder for that specific list element
+            GenomicsDBVidMapProto.GenomicsDBFieldInfo.Builder infoBuilder =
+                    updatedVidMapBuilder.getFieldsBuilder(fieldIdx);
+
+            GenomicsDBVidMapProto.FieldLengthDescriptorComponentPB.Builder lengthDescriptorComponentBuilder =
+                    GenomicsDBVidMapProto.FieldLengthDescriptorComponentPB.newBuilder();
+            lengthDescriptorComponentBuilder.setVariableLengthDescriptor("R");
+            infoBuilder.addLength(lengthDescriptorComponentBuilder.build());
+            lengthDescriptorComponentBuilder.setVariableLengthDescriptor("var"); //ignored - can set anything here
+            infoBuilder.addLength(lengthDescriptorComponentBuilder.build());
+            infoBuilder.addVcfDelimiter("|");
+            infoBuilder.addVcfDelimiter(",");
+
+            if (newCombineOperation.equals("histogram_sum")) {
+                //Each element of the vector is a tuple <float, int>
+                infoBuilder.addType("float");
+                infoBuilder.addType("int");
+                infoBuilder.setVCFFieldCombineOperation("histogram_sum");
+            } else {
+                infoBuilder.setVCFFieldCombineOperation("element_wise_sum");
+                if (newCombineOperation.equals("element_wise_float_sum")) {
```

Comment:
The combine operations should ideally be declared as named `static final` constants, rather than hardcoded Strings.